fox_news_scraper/extract_fn.py

```diff
@@ -0,0 +1,90 @@
+"""A file to extract information from various Fox News RSS feeds."""
+
+import feedparser
+import grequests
+import requests
+from bs4 import BeautifulSoup
+
+
+def fetch_rss_feed(feed_url: str) -> feedparser.FeedParserDict:
+    """Fetches and parsers the RSS feed from the provided URL."""
+
+    feed = feedparser.parse(feed_url)
+    return feed
+
+
+def get_article_content(article_url: str) -> str:
+    """Scrapes the full content of an article from a given URL."""
+
+    try:
+        response = requests.get(article_url, timeout=10)
```

Comment:
I think it would be good to avoid using requests entirely since it is inherently synchronous process. We should async to getting the article_url.